docs/source/debugging.rst

```diff
@@ -0,0 +1,237 @@
+Debugging
+=========
+
+Debugging parallel programs is hard.  Normal debugging tools like logging and
+using ``pdb`` to interact with tracebacks stop working normally when exceptions
+occur in far-away machines or different processes or threads.
+
+Dask has a variety of mechanisms to make this process easier.  Depending on
+your situation some of these approaches may be more appropriate than others.
+
+These approaches are ordered from light weight or easy solutions to more
+involved solutions.
+
+Exceptions
+----------
+
+When a task in your computation fails the standard way of understanding what
+went wrong is to look at the exception and traceback.  Often people do this
+with the ``pdb`` module, IPython ``%debug`` or ``%pdb`` magics, or by just
+looking at the traceback and investigating where in their code the exception
+occurred.
+
+Normally when a computation computes in a separate thread or a different
+machine these approaches break down.  Dask provides a few mechanisms to
+recreate the normal Python debugging experience.
+
+Inspect Exceptions and Tracebacks
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, Dask already copies the exception and traceback wherever they
+occur and reraises that exception locally.  If your task failed with a
```

Comment:
I think this assumes that the exception is serializable, if it doesn't you get an infinite recursion error from the depths of pickle.